src/careamics/models/unet.py

```diff
@@ -233,6 +234,118 @@ def forward(self, *features: List[torch.Tensor]) -> torch.Tensor:
+    in_channels : int, optional
+        Number of input channels, by default 1.
+    depth : int, optional
+        Number of downsamplings, by default 3.
+    num_channels_init : int, optional
+        Number of filters in the first convolution layer, by default 64.
+    use_batch_norm : bool, optional
+        Whether to use batch normalization, by default True.
+    dropout : float, optional
+        Dropout probability, by default 0.0.
+    pool_kernel : int, optional
+        Kernel size of the pooling layers, by default 2.
+    last_activation : Optional[Callable], optional
+        Activation function to use for the last layer, by default None.
+    ind_channels: bool, optional
+        Whether to create independent models to be trained on each channel
+        seperately, by default True.
+    """
+
+    def __init__(
+        self,
+        conv_dims: int,
+        num_classes: int = 1,
+        in_channels: int = 1,
+        depth: int = 3,
+        num_channels_init: int = 64,
+        use_batch_norm: bool = True,
+        dropout: float = 0.0,
+        pool_kernel: int = 2,
+        final_activation: Union[SupportedActivation, str] = SupportedActivation.NONE,
+        n2v2: bool = False,
+        ind_channels: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        conv_dims : int
+            Number of dimensions of the convolution layers (2 or 3).
+        num_classes : int, optional
+            Number of classes to predict, by default 1.
+        in_channels : int, optional
+            Number of input channels, by default 1.
+        depth : int, optional
+            Number of downsamplings, by default 3.
+        num_channels_init : int, optional
+            Number of filters in the first convolution layer, by default 64.
+        use_batch_norm : bool, optional
+            Whether to use batch normalization, by default True.
+        dropout : float, optional
+            Dropout probability, by default 0.0.
+        pool_kernel : int, optional
+            Kernel size of the pooling layers, by default 2.
+        last_activation : Optional[Callable], optional
+            Activation function to use for the last layer, by default None.
+        ind_channels: bool, optional
+            Whether to create independent models to be trained on each channel
+            seperately, by default True.
+        """
+        super().__init__()
+
+        model_kwargs = {
+            "conv_dims": conv_dims,
+            "num_classes": num_classes,
+            "in_channels": in_channels,
+            "depth": depth,
+            "num_channels_init": num_channels_init,
+            "use_batch_norm": use_batch_norm,
+            "dropout": dropout,
+            "pool_kernel": pool_kernel,
+            "final_activation": final_activation,
+        }
+        # TODO: This could be part of the `model_factory` function
```

Comment:
Agreed, that would probably make more sense to have it refactored.

The "`independent_channel` could be a parameter of the `AlgorithmConfig` (it changed name recently) which would be passed on to `model_factory` along the model config (https://github.com/CAREamics/careamics/blob/main/src/careamics/lightning_module.py#L59).

That way we can also pass the name of the in/out channel according to the model to the `ind_channel_model`.

Reply:
I was thinking this too. At the moment it's in the model because it evolved from adapting the UNet architecture. I can start refactoring.